### PR TITLE
added advertisers

### PR DESCRIPTION
**Pull request by nenau-again**
They browse the system and broadcast stupid slogans. Of course, much more can be added...
I made sure they are not too numerous.

### CODE REVIEW

**Comment by agb32** [comment]
Looks good - change denunce to denounce maybe?

File: dat/ai/advertiser.lua [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      standing = ai.getstanding( p ) or -1
+      mem.refuel_msg = string.format(_("\"I'll supply your ship with fuel for %d credits.\""),
+            mem.refuel);
+   end
+
+   -- Selects an advertiser message
+   msg = {
+      _("Fly safe, fly Milspec."),
+      _("Reynir's Hot Dogs: enjoy the authentic taste of tradition."),
+      _("Everyone is faster that light, but only Tricon engines are faster than thought!"),
```
Comment (onpon4) [inline dat/ai/advertiser.lua:56]: that -> than

**Comment by onpon4** [comment]
Is there any particular reason for duplicating civilian.lua rather than just extending that by using `require "dat/ai/civilian.lua"`?

🕵️

File: dat/factions/spawn/civilian.lua [hunk 2]
```diff
@@ -4,15 +4,27 @@ require("dat/factions/spawn/common.lua")
...
-    local civships = {{"Civilian Schroedinger", 8},
-                      {"Civilian Llama", 8},
-                      {"Civilian Gawain", 8},
-                      {"Civilian Hyena", 13}
-                     }
-    
-    local select = rnd.rnd(1, #civships)
+    local r = rnd.rnd()
+
+    if r < 0.5 then
```
Comment (bobbens) [inline dat/factions/spawn/civilian.lua:9]: This should probably be based on presence. I don't think very unsafe (pirates, etc.) systems should have too many advertisers. Should only somewhat exist in systems that are very populated.

File: dat/ai/advertiser.lua [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      _("Dare excellence! Dare Teracom rockets!"),
+      _("Because being unique is hard at times, Nexus invented the Shark fighter."),
+      _("Never take off without your courage. Never take off without your Vendetta."),
+      _("Unicorp: low price and high quality!")
+   }
+
+   -- Faction specific messages
+   if system.cur():presence(faction.get("Empire")) > 1 then
```
Comment (bobbens) [inline dat/ai/advertiser.lua:64]: Seems like a lot of messages are being duplicated. Might be better to try to organize it into a common file.

**Comment by nenau-again** [comment]
>Personally I'm also not sold on 50% of civilians becoming these "advertiser" ships. That seems like far too many; it'd get old pretty quick I think.

Actually, I've been surprised how low the civilian presence is. As a consequence, 50% is what I needed to get a quantity of message comparable to the quantity of imperial ships messages.

>Might be better to try to organize it into a common file.

I don't understand. The messages are already in the same file.

**Comment by bobbens** [comment]
> > Might be better to try to organize it into a common file.
> 
> I don't understand. The messages are already in the same file.

I mean that we have lots of messages duplicated in different files like `dat/ai/civilian.lua` and such. It might be better to finally condense them all into a helper script like `dat/ai/chat.lua` or something so we don't have things like `_("Mayday! We are under attack!")` repeated 4-5 times like we do now ;)

**Comment by onpon4** [comment]
> As a consequence, 50% is what I needed to get a quantity of message comparable to the quantity of imperial ships messages.

That makes sense, but I don't think trying to be as frequent as Empire ships should be the goal. The idea works a lot better if it's a rare occurrence I think.

🕵️

**Comment by nenau-again** [comment]
Allright. I think I've corrected everything, except for the 50% question. I'll let you test and modify if needed.
(If you test in Zied, there will be many messages because there are many civilians in few space, but on other systems, it's fairly sparse)

**Comment by bobbens** [comment]
Looks good, thanks! We can tweak the values as needed.

**Pull request merged**